### PR TITLE
Use gitea logging module for git module

### PR DESCRIPTION
**Pull request by 6543**
remove `log()` func from gogs times and switch to proper logging

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Comment by lunny** [comment]
Have you tested if ssh feature is OK? Why the log left because when run gitea serv/hooks commands, you cannot output errors directly to console.

**Comment by 6543** [comment]
build in ssh works & we can remove this logging but i think it's ok - feel free to add delete suggestions :)

File: modules/git/command.go [hunk 1]
```diff
@@ -114,9 +115,9 @@ func (c *Command) RunInDirTimeoutEnvFullPipelineFunc(env []string, timeout time.
 	}
 
 	if len(dir) == 0 {
-		log(c.String())
+		log.Debug(c.String())
```
Comment (zeripath) [inline modules/git/command.go]:
```suggestion
		log.Debug("%s", c)
```

File: modules/git/command.go [hunk 2]
```diff
@@ -198,9 +199,6 @@ func (c *Command) RunInDirTimeoutEnv(env []string, timeout time.Duration, dir st
 		return nil, ConcatenateError(err, stderr.String())
 	}
 
-	if stdout.Len() > 0 {
-		log("stdout:\n%s", stdout.Bytes()[:1024])
-	}
```
Comment (zeripath) [inline modules/git/command.go]: Could replace this with a trace level:

```suggestion
if stdout.Len() > 0 && log.IsTrace() {
  log.Trace("Stdout:\n %s", stdout.Bytes()[:1024])
}
```

The output of the git command has come in handy occasionally and Trace is probably the level it should come out at.

Comment (zeripath) [reply modules/git/command.go]: (we need the log.IsTrace() test becase we should only bother with sending the message if it could possibly be logged.)

**Comment by zeripath** [comment]
> Have you tested if ssh feature is OK? Why the log left because when run gitea serv/hooks commands, you cannot output errors directly to console.

Ah so my changes here:

https://github.com/go-gitea/gitea/blob/837e8b30a76293fd96d2aae872e30637668ddb3f/cmd/serv.go#L52

Mean that it will work fine.

**Comment by zeripath** [comment]
Ok we need to do the setlogger thing in maintest in git

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Comment by jpraet** [comment]
I get some debug and trace logging at startup even when only info logging is enabled, I think because the LogService is not yet initialized at that point.

https://github.com/go-gitea/gitea/blob/622f1e764c6230023cc1944ad727cd2ad1544b68/modules/git/command.go#L202

This generates these \003d in the logs:

```
2021/06/26 09:58:47 ...dules/git/command.go:118:RunInDirTimeoutEnvFullPipelineFunc() [D] C:\Program Files\Git\mingw64\bin\git.exe version
2021/06/26 09:58:48 ...dules/git/command.go:202:RunInDirTimeoutEnv() [T] Stdout:
         git version 2.18.0.windows.1
        \003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d\003d etc...
```

**Comment by zeripath** [comment]
Don't comment on merged or closed issues and pulls open a new one and reference instead.

I'll take a look though

**Comment by zeripath** [comment]
OK I think we still need the Trace otherwise there's a serious deprecation in logging - however, we're actually printing out mostly from the buffer here instead of the actual output. 

Now we have to think about how we're going to sort out the rest of the logging that happens - it's kinda bad to get this stuff emitted. I guess we should just move the logging initiation higher in the settings hierarchy.

(Realistically a proper dependency management system would have asserted that anyway.)

**Pull request merged**